packages/framework/agent-scheduler/src/scheduler.ts

```diff
@@ -420,8 +422,11 @@ export class AgentSchedulerFactory implements IFluidDataStoreFactory {
 
     public static async createChildInstance(parentContext: IFluidDataStoreContext): Promise<AgentScheduler> {
         const packagePath = [...parentContext.packagePath, AgentSchedulerFactory.type];
-        const router = await parentContext.containerRuntime.createDataStore(packagePath);
-        return requestFluidObject<AgentScheduler>(router, "/");
+        const dataStore = await parentContext.containerRuntime.createDataStore(packagePath);
+        const maybeHandle: FluidObject<IFluidHandle> = (dataStore as any);
+        const handle = await maybeHandle.IFluidHandle?.get();
+        assert(handle instanceof AgentScheduler, "The data store's handle is not an AgentScheduler!");
```

Comment:
```suggestion
        const maybeAgentScheduler = (await maybeHandle.IFluidHandle?.get()).IAgentScheduler;
        assert(maybeAgentScheduler !== undefined, "The data store's handle is not an AgentScheduler!");
```

Reply:
Hmm so the first line would need `as FluidObject<IAgentScheduler>` in the middle, i.e. `const maybeAgentScheduler = (await maybeHandle.IFluidHandle?.get() as FluidObject<IAgentScheduler>).IAgentScheduler;`, right? But this method is supposed to return an actual `AgentScheduler` and not just the interface, so I'd need to cast that in the end.

Reply:
Above in this file (lines 395-399) the function I'm passing to the FluidDataStoreRuntime to initialize its handle object (`AgentScheduler.load()`), returns an `AgentScheduler` directly, so I think that's what we could expect the handle to be.

Reply:
The main point is to avoid using `instanceof` as Vlad mentions in his feedback.  Even if you cast to `AgentScheduler` before returning, using `IAgentScheduler` as the type check for the assertion is probably the safer option here.

Reply:
Ah, yes, that part makes sense 👍 thanks!